### PR TITLE
Resurrect hb_set_invert()

### PR DESCRIPTION
**Pull request by behdad**

### CODE REVIEW

**Comment by behdad** [comment]
Okay, this is ready for review.

**Comment by behdad** [comment]
Umm. Needs tests to be written. Anyone can help with that?

File: src/hb-bit-set-invertible.hh [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+    {
+      if (unlikely (inverted))
+	process (hb_bitwise_or, other);
+      else
+	process (hb_bitwise_and, other); /* Main branch. */
+    }
+    else
+    {
+      if (unlikely (inverted))
+	process (hb_bitwise_ge, other);
```
Comment (garretrieger) [inline src/hb-bit-set-invertible.hh]: I believe this should be lt

File: src/hb-bit-set-invertible.hh [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+      else
+	process (hb_bitwise_le, other);
```
Comment (garretrieger) [inline src/hb-bit-set-invertible.hh]: gt

File: src/hb-bit-set-invertible.hh [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+    {
+      if (unlikely (inverted))
+	process (hb_bitwise_lt, other);
+      else
+	process (hb_bitwise_gt, other); /* Main branch. */
+    }
+    else
+    {
+      if (unlikely (inverted))
+	process (hb_bitwise_non, other);
```
Comment (garretrieger) [inline src/hb-bit-set-invertible.hh]: or

File: src/hb-bit-set-invertible.hh [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+    auto old = *codepoint;
+    auto v = old;
+    s.next (&v);
+    if (old + 1 < v)
+    {
+      *codepoint = old + 1;
+      return true;
+    }
+
+    if (unlikely (old + 1 == INVALID))
```
Comment (garretrieger) [inline src/hb-bit-set-invertible.hh:192]: I think you also want to check if old == INVALID

Comment (behdad) [reply src/hb-bit-set-invertible.hh:192]: No, `old == INVALID` is how we start iteration. `old+1` becomes 0 and it all works.

**Pull request merged**